src/Cli/ConfigGenerator.cs

```diff
@@ -1317,5 +1317,53 @@ private static EntityGraphQLOptions ConstructUpdatedGraphQLDetails(Entity entity
 
             return graphQLType with { Operation = graphQLOperation };
         }
+
+        /// <summary>
+        /// This method will add the telemetry options to the config file. if the config file already has telemetry options,
+        /// it will overwrite the existing options.
+        /// Data API builder consumes the config file with provided telemetry options to send telemetry to Application Insights.
+        /// </summary>
+        public static bool TryAddTelemetry(AddTelemetryOptions options, FileSystemRuntimeConfigLoader loader, IFileSystem fileSystem)
+        {
+            if (!TryGetConfigFileBasedOnCliPrecedence(loader, options.Config, out string runtimeConfigFile))
+            {
+                return false;
+            }
+
+            if (!loader.TryLoadConfig(runtimeConfigFile, out RuntimeConfig? runtimeConfig))
+            {
+                _logger.LogError("Failed to read the config file: {runtimeConfigFile}.", runtimeConfigFile);
+                return false;
+            }
+
+            if (runtimeConfig.Runtime is null)
+            {
+                _logger.LogError("Invalid or missing 'runtime' section in config file: {runtimeConfigFile}.", runtimeConfigFile);
+                return false;
+            }
+
+            if (string.IsNullOrWhiteSpace(options.AppInsightsConnString))
```

Comment:
If we are checking for the validity of the connstring, shouldn't this ensure the connection actually succeeds to the appinsights telemetry resource? Also, we should add this validation to the `dab validate` command.

Reply:
As of now i could only found one way to check whether connection succeds or not is by sending telemetry to App Insights and checking in the portal itself.